sdk/storage/azure-storage-common/src/shared_key_policy.cpp

```diff
@@ -66,13 +66,14 @@ namespace Azure { namespace Storage {
 
     // canonicalized resource
     string_to_sign += "/" + m_credential->AccountName + "/" + request.GetUrl().GetPath() + "\n";
-    for (const auto& query : request.GetUrl().GetQuery())
+    for (const auto& query : request.GetUrl().GetEncodedQuery())
     {
       std::string key = query.first;
       std::transform(key.begin(), key.end(), key.begin(), [](char c) {
         return static_cast<char>(std::tolower(static_cast<unsigned char>(c)));
       });
-      ordered_kv.emplace_back(std::make_pair(std::move(key), query.second));
+      ordered_kv.emplace_back(
+          std::make_pair(std::move(key), Azure::Core::Http::Url::Decode(query.second)));
```

Comment:
We should also decode `key` because in this [API reference](https://docs.microsoft.com/en-us/rest/api/storageservices/authorize-with-shared-key), it says `URL-decode each query parameter name and value.`.

Reply:
Done